src/ui/space-ui.vue

```diff
@@ -191,8 +219,52 @@ export default {
      * @param  {String} item
      * @return {String}
      */
-    componentName(item) {
-      return utils.label(utils.references.getComponentName(item.logicData.component._ref));
+    componentNameFromLogic(logicData) {
+      return utils.label(utils.references.getComponentName(logicData.component._ref));
+    },
+    createReadoutLabel(componentData, property) {
+      return Array.isArray(property)
+              ? property
+                  .slice(0, MAX_PROPERTIES_FOR_READOUT_LABEL)
+                  .map(prop => componentData[prop])
+                  .join(' — ')
+              : '' + componentData[prop]
+    },
+
+
+    /**
+     * Creates readouts for a component. These are shown in the
+     * UI to distinguish component instances based on properties
+     * identified in the `_targeting` field of the component's schema
+     *
+     * @param {Object} componentData has data for each property in schema `_targeting` field
+     * @param {Object} componentSchema with an optional _targeting field
+     * @return {Array<Object<String, String>>} {label, icon} for each target
+     *
+     */
+    createReadouts(componentData, componentSchema) {
```

Comment:
This data cannot be derived from the component's data because a component should be able to live in ANY context, not just in a Space. This would require adding a non-standard value to the schema of any component you are considering for a Space. 

Instead we want to attach this readout logic to the component that actually does the targeting, i.e. the Logic. I'm going to update this implementation as I refactor the icons.